### DISCUSSION TITLE
Is it possible to run cloud-init only on firstboot?

### DISCUSSION THREAD

**Discussion by Ranjandas** [lima-vm/lima#2598] [Q&A]
Is it possible to run cloud-init only on the first boot and not on VM restarts in Lima?

**Comment by afbjorklund** [comment]
While that is the cloud-init default, it is currently not configurable in Lima...

A new instance ID is generated on each reboot, to run the network scripts.

**Reply by Ranjandas** [reply]
Thank you @afbjorklund. Do you think this is worth filing a feature request for? Or is it expected that Lima template scripts should always be idempotent? I don't think it is made obvious for someone new to lima that those provisioner scripts are fed executed inside cloud-init on startup and would run even on restarts (expect for a comment in the default.yaml template I believe).

**Reply by afbjorklund** [reply]
They are supposed to be idempotent (able to run multiple times), but there could still be an option to re-generate the ID and there could be a command (like `provision`) to run the scripts on-demand as well. But documenting it would be a good start!
